OCPBUGS-42627:updates UDN annotation details (#82764)
openshift/openshift-docs: modules/nw-udn-best-practices.adoc

* `openshift-*` namespaces should not be used to set up a UDN.

> * 2 masquerade IPs are required for user defined networks. You must reconfigure your masquerade subnet to be large enough to hold the required number of networks.

Reviewer: Should this say addresses to be more specific?
```suggestion
* 2 masquerade IP addresses are required for user defined networks. You must reconfigure your masquerade subnet to be large enough to hold the required number of networks.
```

+
[IMPORTANT]
====

> For {product-title} 4.17 and later versions, clusters use `169.254.0.0/17` for IPv4 and `fd69::/112` for IPv6 as the default masquerade subnet. These ranges should be avoided by users. For upgraded clusters, there is no change to the default masquerade subnet.

Reviewer: Up to you but I think "versions" can be dropped here:

```suggestion
For {product-title} 4.17 and later, clusters use `169.254.0.0/17` for IPv4 and `fd69::/112` for IPv6 as the default masquerade subnet. These ranges should be avoided by users. For upgraded clusters, there is no change to the default masquerade subnet.
```

> For {product-title} 4.17 and later versions, clusters use `169.254.0.0/17` for IPv4 and `fd69::/112` for IPv6 as the default masquerade subnet. These ranges should be avoided by users. For upgraded clusters, there is no change to the default masquerade subnet.

Reviewer: Update is the preferred term per the [term glossary](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#update):
```suggestion
For {product-title} 4.17 and later versions, clusters use `169.254.0.0/17` for IPv4 and `fd69::/112` for IPv6 as the default masquerade subnet. These ranges should be avoided by users. For updated clusters, there is no change to the default masquerade subnet.
```

> * The cluster subnet and services CIDR for a UDN cannot overlap with the default cluster subnet CIDR. OVN-Kubernetes network plugin uses `100.64.0.0/16` as the default network's join subnet, you must not use that value to configure a UDN `JoinSubnets` field.

Reviewer 2: aren't we using 100.65.0.0/16 for the primary UDN ? 

We should call that out as the default for UDN, and that the customer must override the join subnets if their subnets somehow overlap w/ that one. 

@tssurya to keep me honest here.

Author: I updated with a note that this must be overridden if your cluster's network uses that address anywhere.

Reviewer 3: From what I know, since default join-subnet may change we avoided specifying it

Reviewer 2: But, as a user, how do I know if I need to override the value ??? 

The admin(s) will know the ranges used in their networks, but if don't know what is the value being used here, they have no chance of knowing when they need to define a different join address.
